gpii/node_modules/matchMakerFramework/src/MatchMakerUtilities.js

```diff
@@ -461,4 +461,54 @@ var fluid = fluid || require("infusion"),
             }
         });
     };
+
+    /*
+     * Converts application specific settings into common terms, keyed by application ids. Given a
+     * set of preferences, for each application block in those preferences, the settings for which
+     * inverse transformations exist will be transformed into common terms. All the common terms
+     * that have successfully been inferred from an application will be written to a block keyed by
+     * that applications ID.
+     *
+     * @preferences (Object) - An NP set (including contexts, etc)
+     * @fullSolutionsRegistry (Object) - The solutionsRegistries, keyed by OS
+     *
+     * @return (Object) - Object where keys are solution IDs and entries are common term
+     *      preferences which have been transformed from application specific settings for that
+     *      soluion
+     */
+    gpii.matchMakerFramework.utils.inferCommonTerms = function (preferences, fullSolutionsRegistry) {
+        var flattenedSolutionsRegistry = {};
+        fluid.each(fullSolutionsRegistry, function (entry) {
+            fluid.extend(flattenedSolutionsRegistry, entry);
+        });
+        var togo = {};
+        fluid.each(preferences.contexts, function (context, contextId) {
+            togo[contextId] = {};
+            fluid.each(context.preferences, function (prefsBlock, prefsId) {
+                if (prefsId.indexOf("http://registry.gpii.net/applications/") === 0) {
```

Comment:
Break all of this block out into its own function. Also, we are now in a sufficiently modern world that you can now simply write prefsId.startsWith("http:// " .... etc